### PR TITLE
Fixes #14436 - exception handling for parsing configuration - take 2

### PR DESCRIPTION
**Pull request by tstrachota**
Since the [original PR](https://github.com/theforeman/hammer-cli/pull/206) has been silent for a while I'm opening a new one with with additional commit that should address the comments.

### CODE REVIEW

File: lib/hammer_cli/settings.rb [hunk 1]
```diff
@@ -40,8 +40,8 @@ def self.load_from_file(file_path)
             load(config)
             path_history << file_path
           end
-        rescue Exception
-          puts _("Couldn't load configuration file %s, please check the YAML file for syntax.") % file_path
+        rescue Exception => e
+          puts _("Couldn't load configuration file %{path}: %{message}") % { path: file_path, message: e.message }
```
Comment (mbacovsky) [inline lib/hammer_cli/settings.rb]: After some testing I'd prefer to use warning `warn _("Warning: Couldn't......`.
It feels better visible and more in line with other similar output we do. What do you think?

Comment (tstrachota) [inline lib/hammer_cli/settings.rb]: I agree. In fact I overlooked there's puts instead of warn.

**Comment by tstrachota** [comment]
@mbacovsky updated

File: lib/hammer_cli/settings.rb [hunk 2]
```diff
@@ -34,10 +34,14 @@ def self.load_from_paths(files)
...
-          load(config)
-          path_history << file_path
+        begin
+          config = YAML::load(File.open(file_path))
+          if config
+            load(config)
+            path_history << file_path
+          end
+        rescue Exception => e
+          warn _("Couldn't load configuration file %{path}: %{message}") % { path: file_path, message: e.message }
```
Comment (mbacovsky) [inline lib/hammer_cli/settings.rb]: Almost perfect, could you please prefix the message  with "Warning: " as we do elsewhere.

**Comment by tstrachota** [comment]
...a šmidra vydra, @mbacovsky

**Comment by mbacovsky** [comment]
Significantly better, Bobe! :panda_face: 
:+1:
Will be merged without squashing after checks finish.

**Pull request merged**